src/components/Hoverable/index.js

```diff
@@ -59,6 +69,25 @@ class Hoverable extends Component {
         }
     }
 
+    /**
+     * Toggle scrolling status and set the hover state
+     */
+    setScrollingAndStartUnsetListener() {
+        this.isScrolling = true;
+        this.unsetScrollingAndSetHovered();
+    }
+
+    /**
+     * Unset the scrolling status and set the hover state to true when user is hovering the component.
+     */
+    unsetScrollingAndSetHovered() {
+        this.isScrolling = false;
+        if (!this.hovering) {
+            return;
+        }
+        this.setIsHovered(true);
```

Comment:
**thought:** "hover" is independent from "scrolling". They are two different states. When we hover we set "hover" when we scroll we set "scroll". However, scrolling is unique in that we should prevent "showing" hover while we are scrolling.

So, it make more sense to do something like:

- Set `isScrolling: true` to state in a throttled scroll listener
- When we are not scrolling set `isScrolling: false`
- In `render()` `const hovered = !this.state.isScrolling && this.state.isHovered`

Reply:
Great suggestion. I will try this.

Reply:
I think it would be expensive. This will cause a render on each mounted message item when the user is scrolling.

Reply:
Even if throttled? What if we just set state on the leading and trailing edges of the scroll?